.github/workflows/prototype-tests-gpu.yml

```
@@ -0,0 +1,80 @@
+# prototype-tests.yml adapted for self-hosted with gpu
+name: tests-gpu
+
+on:
+  pull_request:
+
+jobs:
+  prototype:
+    strategy:
+      fail-fast: false
+
+    runs-on: [self-hosted, linux.4xlarge.nvidia.gpu]
+    container:
+      image: pytorch/conda-builder:cuda116
+      options: --gpus all
+
+    steps:
+      - name: Run nvidia-smi
+        run: nvidia-smi
+
+      - name: Upgrade system packages
+        run: python -m pip install --upgrade pip setuptools wheel
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install PyTorch nightly builds
+        run: pip install --progress-bar=off --pre torch torchdata --extra-index-url https://download.pytorch.org/whl/nightly/cu116/
+
+      - name: Install torchvision
+        run: pip install --progress-bar=off --no-build-isolation --editable .
+
+      - name: Install other prototype dependencies
+        run: pip install --progress-bar=off scipy pycocotools h5py iopath
+
+      - name: Install test requirements
+        run: pip install --progress-bar=off pytest pytest-mock pytest-cov
+
+      - name: Mark setup as complete
+        id: setup
+        run: python -c "import torch; print(torch.__version__); print(torch.cuda.is_available())" && exit 0
```

Is this needed? We are explicitly installing from the CU116 wheels above. What extra information are you trying to get here?

---

Well, this just some debug info. If I do not pass `--gpus=all` flag and skip `nvidia-smi` step, then everything will be installed and tests will run without cuda tests and we wont see that...

---

Shouldn't we assert it then? Something like

```suggestion
        run: python -c "import sys, torch; sys.exit(not torch.cuda.is_available())
```

---

yes, if we need to assert that we can do
```bash
python -c "import torch; exit(not torch.cuda.is_available())"
```